test/Microsoft.NET.TestFramework/CoreMSBuildAndWindowsOnlyTheoryAttribute.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.DotNet.PlatformAbstractions;
+using Xunit;
+
+namespace Microsoft.NET.TestFramework
+{
+    public class CoreMSBuildAndWindowsOnlyTheoryAttribute : TheoryAttribute
```

Comment:
Roslyn has a nice pattern for combining common skip conditions: https://github.com/dotnet/roslyn/blob/master/src/Test/Utilities/Portable/Assert/ConditionalFactAttribute.cs

I think we should steal it as it allows composing things like WindowsOnly and CoreMSBuildOnly and also has the benefit of consistency with other dotnet repos.